lib/scholar/metrics/distance.ex

```diff
@@ -240,25 +368,53 @@ defmodule Scholar.Metrics.Distance do
       iex> y = Nx.tensor([1, 2, 3])
       iex> Scholar.Metrics.Distance.cosine(x, y)
       ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+
+      iex> x = Nx.tensor([[1, 2, 3], [0, 0, 0], [5, 2, 4]])
+      iex> y = Nx.tensor([[1, 5, 2], [2, 4, 1], [0, 0, 0]])
+      iex> Scholar.Metrics.Distance.cosine(x, y, axes: [1])
+      #Nx.Tensor<
+        f32[3][3]
+        [
+          [0.1704850196838379, 0.2418246865272522, 1.0],
+          [1.0, 1.0, 0.0],
+          [0.3740193247795105, 0.2843400239944458, 1.0]
+        ]
+      >
```

Comment:
In other cases we just batch multiple distance computations, but in this case we calculate the distance between each pair of sub-tensors, is there a good reason for this?

Reply:
Probably not, I will change that.